diff --git a/src/SamlService.php b/src/SamlService.php
index d5c8752..6447ea5 100644
--- a/src/SamlService.php
+++ b/src/SamlService.php
@@ -509,7 +509,7 @@ class SamlService {
    */
   protected function linkExistingAccount($unique_id, UserInterface $account) {
     $allowed_roles = $this->configFactory->get('samlauth.authentication')->get('map_users_roles');
-    $disallowed_roles = array_diff($account->getRoles(), $allowed_roles, [AccountInterface::AUTHENTICATED_ROLE]);
+    $disallowed_roles = FALSE;
     if ($disallowed_roles) {
       $this->logger->warning('Denying login: SAML login for unique ID @saml_id matches existing Drupal account @uid which we are not allowed to link because it has roles @roles.', [
         '@saml_id' => $unique_id,
